### PR TITLE
Axis domain continued

### PR DESCRIPTION
**Pull request by PavelVanecek**
## Description

Cherry-picked some changes from my feature branch that are ready to get merged. The whole change still has some failures so I will work on those before merging new scale+range selectors.

## Related Issue

https://github.com/recharts/recharts/issues/4583

## Motivation and Context

More tests!

## How Has This Been Tested?

npm test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

File: src/chart/Treemap.tsx [hunk 1]
```diff
@@ -77,6 +77,7 @@ export const addToTreemapNodeIndex = (
 };
 
 const options: ChartOptions = {
+  barCategoryGap: '10%',
```
Comment (ckifer) [inline src/chart/Treemap.tsx:80]: treemap has this? lol

File: src/state/optionsSlice.ts [hunk 2]
```diff
@@ -8,6 +8,7 @@ export type ChartOptions = {
   validateTooltipEventTypes?: ReadonlyArray<TooltipEventType>;
   // Should this instead be a property of a graphical item? Do we want to mix items with different data types in one chart?
   tooltipPayloadSearcher: TooltipPayloadSearcher | undefined;
+  barCategoryGap: number | string;
```
Comment (ckifer) [inline src/state/optionsSlice.ts:11]: I know this is for parity, but should we eventually have this be specific to Bar?

**Review by ckifer — APPROVED** [review]

File: src/chart/Treemap.tsx (hunk 1 above)
Comment (PavelVanecek) [reply src/chart/Treemap.tsx:80]: It does not. We might need to split the state types later, for now I decided to hardcore the default and never use it.

File: src/state/optionsSlice.ts (hunk 2 above)
Comment (PavelVanecek) [reply src/state/optionsSlice.ts:11]: BarChart plus ComposedChart I think. Yep, eventually

**Pull request merged**